### ISSUE TITLE
numsamples parameter for stats.sample() is deprecated

### ISSUE THREAD

**Issue by mostlyaman** [sympy/sympy#21723]
### History of major changes in sample
In SymPy 1.6, `sample` function used the size parameter which is used to specify the no. of samples returned at a time.
```python
>>> N = Normal('N', 10, 2)
>>> sample(N, size = 2)
[11.5448014679636, 11.5448014679636]
```
In SymPy 1.7, 'sample' function behaviour was changed such that it returns an iterator instead of sample values. Since an iterator is returned, a numsamples parameter was introduced which specified the length of the iterator.
```python
>>> x = sample(N, size= 2, numsamples = 1)
>>> next(x)
array([10.10035932, 12.10979044])
>>> next(x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
```
This new behaviour was considered confusing as discussed in #21563. Also, sample_iter was supposed to be used if a iterator was needed making the new sample to be redundant.
Through the work in PR #21590, sample is reverted to its original behaviour in SymPy 1.9.

### Why numsamples is Deprecated for sample
1. Since, sample does not return an iterator now, it does not need a parameter to set the length of the iterator. However, it is still functional although supposed to be removed in a furthur release. If you want to work with an iterator, please use sample_iter instead

2. If numsamples is used with the new sample behaviour, it returns a list of samples with each element of length size.
```python
 >>> sample(N, size=3, numsamples=2)
[array([8.65297477, 8.43609025, 8.43507149]), array([ 8.4551409 , 10.36578991, 10.81603399])]
```
This can also be achieved by adding another dimension to the size parameter. It is to be noted that using numsamples returns a list whereas using size returns a numpy array.
```python
>>> sample(N, size=(2,3))
array([[10.31451261,  9.29506023, 11.09854879],
       [ 5.78358422,  8.51109574,  7.17632097]])
```

In conclusion, numsamples serves no unique function but is still provided with SymPy 1.9 for backwards compatibility. It is advised to use size parameter as demonstrated in point 2 instead of numsamples.

**Comment by praneethratna** [comment]
@oscarbenjamin Hey , I would like to work on fixing this issue?
